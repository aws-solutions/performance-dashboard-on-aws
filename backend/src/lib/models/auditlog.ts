/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

export enum ItemEvent {
    Create = "Create",
    Update = "Update",
    Delete = "Delete",
}

export type ModifiedProperty = {
    property: string;
    oldValue: any;
    newValue: any;
};

export interface AuditLog {
    itemId: string;
    timestamp: Date;
    userId: string;
    event: ItemEvent;
    modifiedProperties?: ModifiedProperty[];
}

// Represents a dynamodb item in the Audit Trail table
export interface AuditLogItem {
    pk: string;
    sk: string;
    type: string;
    userId: string;
    event: ItemEvent;
    modifiedProperties?: ModifiedProperty[];
}

export interface DashboardAuditLog extends AuditLog {
    version: number;
}

// Represents a dynamodb item in the Audit Trail table that
// contains attributes specific to Dashboard audit logs.
export interface DashboardAuditLogItem extends AuditLogItem {
    version: number;
}
