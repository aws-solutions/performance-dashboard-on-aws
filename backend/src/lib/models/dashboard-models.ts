export type Dashboard = {
    id: string,
    name: string,
    topicAreaId: string,
    topicAreaName: string,
    description: string,
    createdBy: string,
};

export type DashboardList = Array<Dashboard>;

export interface DashboardItem {
    pk: string,
    sk: string,
    type: string,
    dashboardName: string,
    topicAreaId: string,
    topicAreaName: string,
    description: string,
    createdBy: string,
};