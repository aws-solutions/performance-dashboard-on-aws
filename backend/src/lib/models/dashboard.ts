/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import { Widget } from "./widget";

export const DASHBOARD_ITEM_TYPE = "Dashboard";

export enum DashboardState {
    Draft = "Draft",
    Published = "Published",
    Archived = "Archived",
    /**
     * @type {string}
     * @deprecated in favor of {@link Draft}
     */
    PublishPending = "PublishPending",
    Inactive = "Inactive",
}

export interface DashboardVersion {
    id: string;
    version: number;
    state: DashboardState;
    friendlyURL?: string;
}

export interface Dashboard {
    id: string;
    name: string;
    version: number;
    parentDashboardId: string;
    topicAreaId: string;
    topicAreaName: string;
    displayTableOfContents: boolean;
    /**
     * @type {any}
     * @deprecated no longer supported
     */
    tableOfContents?: any;
    description: string;
    createdBy: string;
    updatedAt: Date;
    updatedBy?: string;
    submittedBy?: string;
    publishedBy?: string;
    archivedBy?: string;
    deletedBy?: string;
    state: DashboardState;
    releaseNotes?: string;
    widgets?: Array<Widget>;
    friendlyURL?: string;
}

export type DashboardList = Array<Dashboard>;

export interface DashboardItem {
    pk: string;
    sk: string;
    type: string;
    version: number;
    parentDashboardId: string;
    dashboardName: string;
    topicAreaId: string;
    topicAreaName: string;
    displayTableOfContents: boolean;
    tableOfContents?: any;
    description: string;
    createdBy: string;
    updatedAt: string;
    updatedBy?: string;
    submittedBy?: string;
    publishedBy?: string;
    archivedBy?: string;
    deletedBy?: string;
    state: string;
    releaseNotes?: string;
    friendlyURL?: string;
}

// Public representation of a dashboard. Hides some fields
// like createdBy and state, which should not be exposed
// on public endpoints.
export interface PublicDashboard {
    id: string;
    name: string;
    topicAreaId: string;
    topicAreaName: string;
    displayTableOfContents: boolean;
    tableOfContents?: any;
    description: string;
    updatedAt: Date;
    widgets?: Array<Widget>;
    friendlyURL?: string;

    // Filled on a homepage search with dashboard content
    // that matches the search query.
    queryMatches?: Array<string>;
}
