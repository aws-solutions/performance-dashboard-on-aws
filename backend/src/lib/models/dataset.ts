/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

export enum SourceType {
    IngestApi = "IngestApi",
    FileUpload = "FileUpload",
}

export enum DatasetSchema {
    None = "None",
    Metrics = "Metrics",
}

export interface Dataset {
    id: string;
    createdBy: string;
    fileName: string;
    s3Key: {
        raw: string;
        json: string;
    };
    updatedAt: Date;
    sourceType: SourceType;
    schema?: DatasetSchema;
}

export type DatasetContent = Array<object>;
export type DatasetList = Array<Dataset>;

export interface DatasetItem {
    pk: string;
    sk: string;
    type: string;
    createdBy: string;
    fileName: string;
    s3Key: {
        raw: string;
        json: string;
    };
    updatedAt: string;
    sourceType: string;
    schema?: string;
}
