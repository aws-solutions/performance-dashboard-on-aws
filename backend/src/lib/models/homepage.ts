/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

export interface Homepage {
    title: string;
    description: string;
    updatedAt: Date;
}

export interface HomepageItem {
    pk: string;
    sk: string;
    type: string;
    updatedAt?: string;
    title: string;
    description: string;
}
