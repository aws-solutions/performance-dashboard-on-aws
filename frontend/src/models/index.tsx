/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

export type TopicArea = {
    id: string;
    name: string;
    createdBy: string;
    dashboardCount: number;
};

export const TopicAreaSortingCriteria = (a: TopicArea, b: TopicArea) => (a.name > b.name ? 1 : -1);

export type PublicTopicArea = {
    id: string;
    name: string;
    dashboards?: Array<Dashboard | PublicDashboard>;
};

export enum DashboardState {
    Draft = "Draft",
    Published = "Published",
    Archived = "Archived",
    PublishPending = "PublishPending",
    Inactive = "Inactive",
}

export enum SourceType {
    IngestApi = "IngestApi",
    FileUpload = "FileUpload",
}

export enum DatasetType {
    DynamicDataset = "DynamicDataset",
    StaticDataset = "StaticDataset",
    CsvFileUpload = "CsvFileUpload",
    CreateNew = "CreateNew",
}

export type Dashboard = {
    id: string;
    name: string;
    version: number;
    parentDashboardId: string;
    topicAreaId: string;
    topicAreaName: string;
    displayTableOfContents: boolean;
    /**
     * @type {any}
     * @deprecated no longer supported
     */
    tableOfContents?: any;
    description?: string;
    releaseNotes?: string;
    widgets: Array<Widget>;
    state: string;
    updatedAt: Date;
    createdBy: string;
    submittedBy?: string;
    publishedBy?: string;
    archivedBy?: string;
    friendlyURL?: string;
    queryMatches?: Array<string>;
};

export type PublicDashboard = {
    id: string;
    name: string;
    topicAreaId: string;
    topicAreaName: string;
    displayTableOfContents: boolean;
    /**
     * @type {any}
     * @deprecated no longer supported
     */
    tableOfContents?: any;
    description?: string;
    widgets: Array<Widget>;
    updatedAt: Date;
    friendlyURL?: string;
    queryMatches?: Array<string>;
};

export type DashboardVersion = {
    id: string;
    version: number;
    state: DashboardState;
    friendlyURL?: string;
};

export type DashboardAuditLog = {
    itemId: string;
    timestamp: Date;
    version: number;
    userId: string;
    event: "Create" | "Update" | "Delete";
    modifiedProperties?: Array<{
        property: string;
        oldValue: any;
        newValue: any;
    }>;
};

export enum WidgetType {
    Text = "Text",
    Chart = "Chart",
    Table = "Table",
    Image = "Image",
    Metrics = "Metrics",
    Section = "Section",
}

export enum ChartType {
    LineChart = "LineChart",
    ColumnChart = "ColumnChart",
    BarChart = "BarChart",
    PartWholeChart = "PartWholeChart",
    PieChart = "PieChart",
    DonutChart = "DonutChart",
}

export interface Widget {
    id: string;
    name: string;
    widgetType: string;
    order: number;
    updatedAt: Date;
    dashboardId: string;
    content: any;
    showTitle: boolean;
    section?: string;
}

export interface ChartWidget extends Widget {
    content: {
        title: string;
        chartType: ChartType;
        datasetId: string;
        summary: string;
        summaryBelow: boolean;
        datasetType?: DatasetType;
        s3Key: {
            raw: string;
            json: string;
        };
        columnsMetadata: ColumnMetadata[];
        sortByColumn?: string;
        sortByDesc?: boolean;
        horizontalScroll?: boolean;
        stackedChart?: boolean;
        significantDigitLabels: boolean;
        dataLabels: boolean;
        computePercentages: boolean;
        showTotal: boolean;
    };
}

export interface TableWidget extends Widget {
    content: {
        title: string;
        datasetId: string;
        summary: string;
        summaryBelow: boolean;
        datasetType?: DatasetType;
        s3Key: {
            raw: string;
            json: string;
        };
        columnsMetadata: ColumnMetadata[];
        sortByColumn?: string;
        sortByDesc?: boolean;
        significantDigitLabels: boolean;
        displayWithPages: boolean;
    };
}

export interface ImageWidget extends Widget {
    content: {
        title: string;
        imageAltText: string;
        summary?: string;
        summaryBelow: boolean;
        s3Key: {
            raw: string;
        };
        filename: string;
        scalePct?: string;
    };
}

export interface MetricsWidget extends Widget {
    content: {
        title: string;
        datasetId: string;
        oneMetricPerRow: boolean;
        datasetType?: DatasetType;
        significantDigitLabels: boolean;
        metricsCenterAlign: boolean;
        s3Key: {
            raw: string;
            json: string;
        };
    };
}

export interface SectionWidget extends Widget {
    content: {
        title: string;
        summary: string;
        widgetIds?: Array<string>;
        showWithTabs: boolean;
        horizontally?: boolean;
    };
}

export enum DatasetSchema {
    None = "None",
    Metrics = "Metrics",
}

export type Dataset = {
    id: string;
    fileName: string;
    s3Key: {
        raw: string;
        json: string;
    };
    createdBy?: string;
    updatedAt?: Date;
    sourceType?: SourceType;
    schema?: DatasetSchema;
};

export type PublicHomepage = {
    title: string;
    description: string;
    dashboards: Array<PublicDashboard>;
};

export type Homepage = {
    title: string;
    description: string;
    updatedAt: Date;
};

export type PublicSettings = {
    dateTimeFormat: {
        date: string;
        time: string;
    };
    navbarTitle?: string;
    colors: {
        primary: string;
        secondary: string;
    };
    topicAreaLabels?: {
        singular: string;
        plural: string;
    };
    customLogoS3Key?: string;
    customFaviconS3Key?: string;
    contactEmailAddress?: string;
    contactUsContent?: string;
    analyticsTrackingId?: string;
};

export type Settings = {
    publishingGuidance: string;
    dateTimeFormat: {
        date: string;
        time: string;
    };
    updatedAt?: Date;
    navbarTitle?: string;
    colors: {
        primary: string;
        secondary: string;
    };
    topicAreaLabels: {
        singular: string;
        plural: string;
    };
    customLogoS3Key?: string;
    customLogoAltText?: string;
    customFaviconS3Key?: string;
    contactEmailAddress?: string;
    adminContactEmailAddress?: string;
    contactUsContent?: string;
    analyticsTrackingId?: string;
};

export type Metric = {
    title: string;
    value: number;
    changeOverTime?: string;
    percentage?: string;
    currency?: string;
    startDate?: string;
    endDate?: string;
};

// Type for the History object in react-router. Defines the
// location state that is common across all routes.
// Usage in a component: useHistory<LocationState>();
export type LocationState = {
    alert?: {
        type: "success" | "warning" | "info" | "error";
        message: string;
        to?: string;
        linkLabel?: string;
    };
    emails?: string;
    usernames?: Array<string>;
    id?: string;
    metrics?: Array<Metric>;
    metric?: Metric;
    position?: number;
    showTitle?: boolean;
    oneMetricPerRow?: boolean;
    significantDigitLabels?: boolean;
    metricsCenterAlign?: boolean;
    metricTitle?: string;
    origin?: string;
    json?: Array<any>;
    staticDataset?: Dataset;
    redirectUrl?: string;
    crumbLabel?: string;
    datasetType?: DatasetType;
};

export enum UserRoles {
    Admin = "Admin",
    Editor = "Editor",
    /**
     * @type {string}
     * @deprecated no longer supported
     */
    Publisher = "Publisher",
    Public = "Public",
}

export type User = {
    userId: string;
    enabled: boolean;
    userStatus: string;
    sub: string;
    email: string;
    roles: Array<UserRoles>;
    createdAt: Date;
    updatedAt: Date;
};

export enum ColumnDataType {
    Number = "Number",
    Text = "Text",
    Date = "Date",
}

export enum NumberDataType {
    Percentage = "Percentage",
    Currency = "Currency",
    "With thousands separators" = "With thousands separators",
}

export enum CurrencyDataType {
    "Dollar $" = "Dollar $",
    "Euro €" = "Euro €",
    "Pound £" = "Pound £",
}

export interface ColumnMetadata {
    columnName: string;
    dataType?: ColumnDataType;
    numberType?: NumberDataType;
    currencyType?: CurrencyDataType;
    hidden: boolean;
}
