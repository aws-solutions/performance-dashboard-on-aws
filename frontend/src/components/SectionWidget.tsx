/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import MarkdownRender from "./MarkdownRender";
import { Widget } from "../models";
import WidgetRender from "./WidgetRender";
import Tabs from "./Tabs";
import TabsVertical from "./TabsVertical";
import { useWindowSize } from "../hooks";
import { Waypoint } from "react-waypoint";
import ShareButton from "./ShareButton";
import Utils from "../services/UtilsService";

interface Props {
    widget: Widget;
    showMobilePreview?: boolean;
    widgets?: Array<Widget>;
    setActiveWidgetId?: Function;
    topOffset?: string;
    bottomOffset?: string;
    defaultActive?: string;
}

function SectionWidget(props: Props) {
    const { content, showTitle } = props.widget;
    const windowSize = useWindowSize();

    let activeTabId = "0";
    if (props.widgets && props.defaultActive) {
        const widget = props.widgets?.find((w: Widget) => w.id === props.defaultActive);
        if (widget && widget.section) {
            const parent = props.widgets?.find((w: Widget) => w.id === widget.section);
            if (parent && parent.id === props.widget.id) {
                const index = parent.content.widgetIds.findIndex(
                    (w: string) => w === props.defaultActive,
                );
                if (index >= 0) {
                    activeTabId = index.toString();
                }
            }
        }
    }

    const chartId = `chart-${Utils.getShorterId(props.widget.id)}`;
    const sectionHeaderId = `section-header-${props.widget.id}`;
    return (
        <div aria-label={content.title}>
            {!content.showWithTabs ? (
                <Waypoint
                    onEnter={() => {
                        if (props.setActiveWidgetId) {
                            props.setActiveWidgetId(props.widget.id);
                        }
                    }}
                    topOffset={props.topOffset}
                    bottomOffset={props.bottomOffset}
                    fireOnRapidScroll={false}
                >
                    <div>
                        {showTitle && (
                            <h2>
                                <span>
                                    {content.title}
                                    <ShareButton
                                        id={`${chartId}a`}
                                        title={content.title}
                                        className="margin-left-1"
                                    />
                                </span>
                            </h2>
                        )}
                        {content.summary ? (
                            <div className="padding-left-05">
                                <MarkdownRender
                                    className="usa-prose textOrSummary"
                                    source={content.summary}
                                />
                            </div>
                        ) : (
                            ""
                        )}
                    </div>
                </Waypoint>
            ) : (
                <>
                    {showTitle && (
                        <h2 id={sectionHeaderId}>
                            {content.title}
                            <ShareButton
                                id={`${chartId}a`}
                                title={content.title}
                                className="margin-left-1"
                            />
                        </h2>
                    )}
                    {content.summary ? (
                        <div className="padding-left-05">
                            <MarkdownRender
                                className="usa-prose textOrSummary"
                                source={content.summary}
                            />
                        </div>
                    ) : (
                        ""
                    )}
                </>
            )}
            {!content.showWithTabs &&
                content.widgetIds &&
                content.widgetIds.map((id: string, index: number) => {
                    const widget = props.widgets?.find((w) => w.id === id);
                    if (widget) {
                        return (
                            <div key={index}>
                                <Waypoint
                                    onEnter={() => {
                                        if (props.setActiveWidgetId) {
                                            props.setActiveWidgetId(widget.id);
                                        }
                                    }}
                                    topOffset={props.topOffset}
                                    bottomOffset={props.bottomOffset}
                                    fireOnRapidScroll={false}
                                >
                                    <div className="margin-top-4 usa-prose" id={id}>
                                        <h3 className="margin-bottom-1">
                                            {widget.content.title}
                                            <ShareButton
                                                id={`${Utils.getShorterId(widget.id)}a`}
                                                title={content.title}
                                                className="margin-left-1"
                                            />
                                        </h3>
                                        <WidgetRender
                                            widget={widget}
                                            showMobilePreview={props.showMobilePreview}
                                            bottomOffset={props.bottomOffset}
                                            topOffset={props.topOffset}
                                            disableShare={true}
                                            hideTitle={true}
                                        />
                                    </div>
                                </Waypoint>
                            </div>
                        );
                    }
                    return false;
                })}
            {content.showWithTabs &&
                (content.horizontally || props.showMobilePreview || windowSize.width <= 600) &&
                content.widgetIds?.length > 0 && (
                    <Tabs defaultActive={activeTabId} showArrows ariaLabelledById={sectionHeaderId}>
                        {content.widgetIds.map((id: string, index: number) => {
                            const widget = props.widgets?.find((w) => w.id === id);
                            if (widget) {
                                return (
                                    <div key={index} id={`${index}`} label={widget.name}>
                                        <WidgetRender
                                            widget={widget}
                                            showMobilePreview={props.showMobilePreview}
                                            hideTitle={true}
                                            disableShare={true}
                                        />
                                    </div>
                                );
                            }
                            return false;
                        })}
                    </Tabs>
                )}
            {content.showWithTabs &&
                !content.horizontally &&
                !props.showMobilePreview &&
                windowSize.width > 600 &&
                content.widgetIds && (
                    <TabsVertical defaultActive={activeTabId} ariaLabelledById={sectionHeaderId}>
                        {content.widgetIds.map((id: string, index: number) => {
                            const widget = props.widgets?.find((w) => w.id === id);
                            if (widget) {
                                return (
                                    <div key={index} id={`${index}`} label={widget.name}>
                                        <WidgetRender
                                            widget={widget}
                                            showMobilePreview={props.showMobilePreview}
                                            hideTitle={true}
                                            disableShare={true}
                                        />
                                    </div>
                                );
                            }
                            return false;
                        })}
                    </TabsVertical>
                )}
        </div>
    );
}

export default SectionWidget;
