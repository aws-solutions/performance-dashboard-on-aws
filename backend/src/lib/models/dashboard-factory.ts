import { v4 as uuidv4 } from "uuid";
import { Dashboard, DashboardItem } from "./dashboard-models";
import { User } from "./user-models";
import topicareaFactory from "./topicarea-factory";

const DASHBOARD: string = 'Dashboard';

function createNew(name: string, topicAreaId: string, topicAreaName: string, user: User): Dashboard {
  return {
    id: uuidv4(),
    name,
    topicAreaId,
    topicAreaName,
    createdBy: user.userId,
  };
}

/**
 * Converts a Dashboard to a DynamoDB item.
 */
function toItem(dashboard: Dashboard): DashboardItem {
  let item: DashboardItem = {
    pk: itemId(dashboard.id),
    sk: itemId(dashboard.id),
    type: DASHBOARD,
    dashboardName: dashboard.name,
    topicAreaName: dashboard.topicAreaName,
    topicAreaId: topicareaFactory.itemId(dashboard.topicAreaId),
    description: dashboard.description,
    overview: dashboard.overview,
    createdBy: dashboard.createdBy,
  };
  return item;
}

/**
 * Converts a DynamoDB item into a Dashboard object
 */
function fromItem(item: DashboardItem): Dashboard {
  const id = item.pk.substring(10);
  let dashboard: Dashboard = {
    id,
    name: item.dashboardName,
    topicAreaId: item.topicAreaId.substring(10),
    topicAreaName: item.topicAreaName,
    description: item.description,
    overview: item.overview,
    createdBy: item.createdBy,
  }
  return dashboard;
}

function itemId(id: string): string { return `${DASHBOARD}#${id}` }

export default {
  toItem,
  createNew,
  fromItem,
  itemId,
};
