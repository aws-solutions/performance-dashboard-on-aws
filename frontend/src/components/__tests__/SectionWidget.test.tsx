/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import SectionWidget from "../SectionWidget";
import { Widget, WidgetType } from "../../models";

const widget: Widget = {
    id: "abc",
    name: "Benefits of Bananas",
    dashboardId: "123",
    order: 1,
    showTitle: true,
    updatedAt: new Date(),
    widgetType: WidgetType.Section,
    content: {
        showWithTabs: false,
        title: "test title",
        summary: "test summary",
    },
};

test("renders the section title and summary", async () => {
    render(
        <SectionWidget
            widget={widget}
            showMobilePreview={false}
            widgets={[widget]}
            setActiveWidgetId={() => {}}
        />,
        { wrapper: MemoryRouter },
    );

    expect(screen.getByText("test title")).toBeInTheDocument();
    const summary = screen.getByText("test summary");
    expect(summary).toBeInTheDocument();
});

test("section preview should match snapshot", async () => {
    const wrapper = render(
        <SectionWidget
            widget={widget}
            showMobilePreview={false}
            widgets={[widget]}
            setActiveWidgetId={() => {}}
        />,
        { wrapper: MemoryRouter },
    );
    expect(wrapper.container).toMatchSnapshot();
});
