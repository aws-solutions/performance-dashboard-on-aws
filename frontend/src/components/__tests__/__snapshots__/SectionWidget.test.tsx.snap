// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`section preview should match snapshot 1`] = `
<div>
  <div
    aria-label="test title"
  >
    <div>
      <h2>
        test title
      </h2>
      <div
        class="padding-left-05"
      >
        <div
          class="Markdown usa-prose textOrSummary"
        >
          <p>
            test summary
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
